packages/webapp/src/containers/Finances/RevenueDetail/index.jsx

```diff
@@ -0,0 +1,99 @@
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect } from 'react';
+import { deleteSale, updateSale } from '../actions';
+import { revenueByIdSelector } from '../selectors';
+import { revenueTypeByIdSelector } from '../../revenueTypeSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { useCurrencySymbol } from '../../hooks/useCurrencySymbol';
+import { setPersistedPaths } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
+import GeneralRevenue from '../../../components/Forms/GeneralRevenue';
+import useCropSaleInputs, { getCustomFormChildrenDefaultValues } from '../useCropSaleInputs';
+import useHookFormPersist from '../../hooks/useHookFormPersist';
+import { revenueTypeTileContentsSelector } from '../../revenueTypeSlice';
+import { mapRevenueFormDataToApiCallFormat, mapRevenueTypesToReactSelectOptions } from '../util';
+
+function RevenueDetail({ history, match }) {
+  const isEditing = match.path.endsWith('/edit');
+  const { sale_id } = match.params;
+
+  // To clear form history after editing
+  useHookFormPersist();
+  const { t } = useTranslation(['translation', 'revenue']);
+  const dispatch = useDispatch();
+
+  const revenueTypes = useSelector(revenueTypeTileContentsSelector);
+  const sale = useSelector(revenueByIdSelector(sale_id));
+  const revenueType = useSelector(revenueTypeByIdSelector(sale?.revenue_type_id));
+
+  useEffect(() => {
+    if (!sale) {
+      history.replace('/unknown_record');
+    }
+  }, [sale, history]);
+
+  // Dropdown should include the current revenue's type even if it has been retired
+  const revenueTypesArray = revenueTypes?.concat(revenueType?.deleted ? revenueType : []);
+  const revenueTypeReactSelectOptions = mapRevenueTypesToReactSelectOptions(revenueTypesArray);
+
+  const onSubmit = (data) => {
+    const editedSale = mapRevenueFormDataToApiCallFormat(data, revenueTypes, sale_id, null);
+    dispatch(updateSale(editedSale));
+  };
+
+  const handleEdit = () => {
+    dispatch(setPersistedPaths([`/revenue/${sale_id}/edit`]));
+    history.push(`/revenue/${sale_id}/edit`);
+  };
+
+  const onRetire = () => {
+    dispatch(deleteSale(sale_id));
+  };
+
+  const handleGoBack = () => {
+    history.back();
+  };
+
+  const onTypeChange = (typeId, setValue, REVENUE_TYPE_OPTION) => {
+    const newType = revenueTypeReactSelectOptions?.find((option) => option.value === typeId);
+    setValue(REVENUE_TYPE_OPTION, newType);
+  };
+
+  return (
+    <GeneralRevenue
+      key={isEditing ? 'editing' : 'readonly'}
+      onSubmit={isEditing ? onSubmit : undefined}
+      title={t('SALE.DETAIL.TITLE')}
```

Comment:
Well no, not quite like that because now both views show the `readonly` title. It needs a conditional:

```suggestion
title={isEditing ? t('SALE.EDIT_SALE.TITLE') : t('SALE.DETAIL.TITLE')}
```

Reply:
(And `EDIT_SALE` needs the wording update if this is the point everything is moving to revenue).

Reply:
"Sale detail" applies to both no? but sure haha

Reply:
I guess it could, yeah 🤷 But that's how it's used elsewhere so I'm pleased by this consistency.